File: scraper/check_dates.py
import os
from supabase import create_client, Client

url: str = os.environ.get("VITE_SUPABASE_URL", "https://bsmrjnviyznzgxzmwgtx.supabase.co")
key: str = os.environ.get("VITE_SUPABASE_ANON_KEY", "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImJzbXJqbnZpeXpuemd4em13Z3R4Iiwicm9sZSI6ImFub24iLCJpYXQiOjE3ODE1NDM0MTIsImV4cCI6MjA5NzExOTQxMn0.4VeOHO42jQHlCivhcctZAkanDPfHDSnjQRnv4Zom4TE")
supabase: Client = create_client(url, key)

response = supabase.table("events").select("id, title, start_date").execute()
events = response.data

if not events:
    print("No events found in DB.")
else:
    print(f"Total events in DB: {len(events)}")
    print("Sample dates:")
    for e in events[:20]:
        print(f" - {e['start_date']} | {e['title']}")

    from datetime import datetime, timedelta, timezone
    now = datetime.now(timezone.utc)
    ten_days_ago = now - timedelta(days=10)
    
    valid_count = sum(1 for e in events if e.get('start_date') is None or (e.get('start_date') and datetime.fromisoformat(e['start_date'].replace('Z', '+00:00')) >= ten_days_ago))
    print(f"\nEvents passing strict 10-day filter: {valid_count} / {len(events)}")

File: backend/check_dates.js
require('dotenv').config();
const { createClient } = require('@supabase/supabase-js');

const supabaseUrl = process.env.VITE_SUPABASE_URL;
const supabaseKey = process.env.VITE_SUPABASE_ANON_KEY;
const supabase = createClient(supabaseUrl, supabaseKey);

async function check() {
    const { data: events, error } = await supabase.from('events').select('id, title, start_date').order('created_at', { ascending: false });
    
    if (error) {
        console.error("DB Error:", error);
        return;
    }

    if (!events || events.length === 0) {
        console.log("No events found in DB.");
        return;
    }

    console.log(`Total events in DB: ${events.length}`);
    console.log("Sample of 20 most recently scraped events:");
    for (let i = 0; i < Math.min(20, events.length); i++) {
        console.log(` - ${events[i].start_date} | ${events[i].title}`);
    }

    const tenDaysAgo = new Date();
    tenDaysAgo.setDate(tenDaysAgo.getDate() - 10);
    
    const valid_count = events.filter(e => {
        if (!e.start_date) return true;
        const d = new Date(e.start_date);
        return d >= tenDaysAgo;
    }).length;
    
    console.log(`\nEvents passing strict 10-day filter: ${valid_count} / ${events.length}`);
}

check();
